src/asm/parser.y

```diff
@@ -694,6 +696,22 @@ endc		: T_POP_ENDC {
 		}
 ;
 
+def_id		: T_OP_DEF {
```

Comment:
I think that this could be "either `DEF <ID>` or `<LABEL>`", to avoid duplicating all symbol definition productions.

It *might* cause parser conflicts, though? And maybe it doesn't make sense, and we'd better stick to this? I'll let you judge, I don't have a strong opinion.

Reply:
The new `def_id EQU/EQUS/SET/etc` directives fall under the `| label directive` pattern of `plain_directive`s, whereas old-style `label EQU/EQUS/SET/etc` are `assignment_directive`s. Also, in future the old-style ones will act differently, defining the constant but also printing a deprecation warning.

Reply:
We'll see whether we deprecate the old syntax; it should still be useful for compatibility, doesn't cost anything, and the new syntax is there to avoid the newbie trap.